modules/exploits/multi/vnc/vnc_keyboard_exec.rb

```diff
@@ -70,8 +70,14 @@ def release_key(key)
 
 
   def exec_command(command)
+    count = 0
     values = command.chars.to_a
     values.each do |value|
+      count += 1
+      if count % 50 == 0
+        # Small sleeps in long commands to avoid overloading keyboard buffer
+        select(nil, nil, nil, 0.05) # 50ms
+      end
```

Comment:
This should probably use the normal `sleep` function instead of select. Metasploit actually uses a [thread safe version](https://github.com/rapid7/metasploit-framework/blob/cb9e4121381430a08fd4b913a14347cd62bc6bec/lib/rex.rb#L102) provided by Rex.

Also if you use `#each_with_index` you wouldn't need to initialize and increment the counter yourself.

```suggestion
    values = command.chars.to_a
    values.each_with_index do |value, index|
      # Small sleeps in long commands to avoid overloading keyboard buffer
      sleep(0.05) if index % 50 == 0
```

Everything else looks good, and we can probably make that change ourselves.

Reply:
Thanks for your feedback! I'm totally not familiar with Ruby so I was not aware of `each_with_index`. I used select() because the module was doing so but sleep() seems better indeed. Applied your code suggestion.